api/context.go

```diff
@@ -299,6 +314,39 @@ func (c *Context) SetUnknownError(where string, details string) {
 	c.Err = model.NewAppError(where, "An unknown error has occured. Please contact support.", details)
 }
 
+func (c *Context) setTeamURL(url string, valid bool) {
+	c.teamURL = url
+	c.teamURLValid = valid
+}
+
+func (c *Context) setTeamURLFromSession() {
```

Comment:
If this does not block, GetTeamURL will print the debug message if it wins the race.